core/src/main/java/com/sequenceiq/cloudbreak/converter/v4/stacks/cluster/CloudStorageConverter.java

```diff
@@ -139,6 +143,7 @@ public FileSystem requestToAdditionalFileSystem(CloudStorageBase cloudStorageReq
         configurations.put(CloudEfsConfiguration.KEY_PERFORMANCE_MODE, efsParameters.getPerformanceMode());
         configurations.put(CloudEfsConfiguration.KEY_PROVISIONED_THROUGHPUT_INMIBPS, efsParameters.getProvisionedThroughputInMibps());
         configurations.put(CloudEfsConfiguration.KEY_THROUGHPUT_MODE, efsParameters.getThroughputMode());
+        configurations.put(CloudEfsConfiguration.KEY_ASSO_INSTANCE_GROUP_NAMES, efsParameters.getAssociatedInstanceGroupNames());
```

Comment:
Where is `getAssociatedInstanceGroupNames()` gets validated? I have seen a null/emptiness check in `AwsEfsFileSystem`, but it is probably too late to do that there. In particular, we shall check for the total number of SGs involved in the specified instancegroups, and freak out if the count exceeds the limit of 5.

Reply:
KEY_ASSO_INSTANCE_GROUP_NAMES -> KEY_ASSOCIATED_INSTANCE_GROUP_NAMES

Reply:
I have added the validation in com.sequenceiq.common.api.cloudstorage.validation.CloudStorageValidator.
changed the key name to KEY_ASSOCIATED_INSTANCE_GROUP_NAMES